src/Aspire.Hosting/Dcp/DcpDependencyCheck.cs

```diff
@@ -52,34 +52,45 @@ internal sealed partial class DcpDependencyCheck : IDcpDependencyCheckService
 
             IAsyncDisposable? processDisposable = null;
             Task<ProcessResult> task;
+            var outputStringBuilder = new StringBuilder();
+            var errorStringBuilder = new StringBuilder();
 
             try
             {
-                var outputStringBuilder = new StringBuilder();
-
                 var arguments = "info";
                 if (!string.IsNullOrEmpty(containerRuntime))
                 {
                     arguments += $" --container-runtime {containerRuntime}";
                 }
 
-                // Run `dcp version`
                 var processSpec = new ProcessSpec(dcpPath)
                 {
                     Arguments = arguments,
                     OnOutputData = s => outputStringBuilder.Append(s),
+                    OnErrorData = s => errorStringBuilder.Append(s),
+                    ThrowOnNonZeroReturnCode = false
                 };
 
                 (task, processDisposable) = ProcessUtil.Run(processSpec);
+                ProcessResult processResult;
 
                 // Disable timeout if DependencyCheckTimeout is set to zero or a negative value
                 if (_dcpOptions.DependencyCheckTimeout > 0)
                 {
-                    await task.WaitAsync(TimeSpan.FromSeconds(_dcpOptions.DependencyCheckTimeout), cancellationToken).ConfigureAwait(false);
+                    processResult = await task.WaitAsync(TimeSpan.FromSeconds(_dcpOptions.DependencyCheckTimeout), cancellationToken).ConfigureAwait(false);
                 }
                 else
                 {
-                    await task.WaitAsync(cancellationToken).ConfigureAwait(false);
+                    processResult = await task.WaitAsync(cancellationToken).ConfigureAwait(false);
+                }
+
+                if (processResult.ExitCode != 0)
+                {
+                    throw new DistributedApplicationException(string.Format(
```

Comment:
The actual exception from the initial issue was actually throwing from the call to `DisposeAsync` in the finally block. The cancellation token had fired, triggering a cancellation exception, but the finally block rethrew a separate exception due to the non-zero exit code from stopping the process.

I think we need to add a try/catch block around that `DisposeAsync` call and re-throw (or wrap and re-throw) the original exception if there is one.

Reply:
Good point. Added a suppression of exceptions from the call to `DisposeAsync()` to ensure that original exception propagates.